Library/ConnectionStringParser.ts

```diff
@@ -0,0 +1,23 @@
+import { IConnectionStringFields } from "../Declarations/Contracts";
+
+class ConnectionStringParser {
+    private static _FIELDS_SEPARATOR = ";";
+    private static _FIELD_KEY_VALUE_SEPARATOR = "=";
+
+    public static parse(connectionString: string): IConnectionStringFields {
+        const kvPairs = connectionString.split(ConnectionStringParser._FIELDS_SEPARATOR);
+
+        return kvPairs.reduce((fields: IConnectionStringFields, kv: string) => {
+            const equalsIndex = kv.indexOf(ConnectionStringParser._FIELD_KEY_VALUE_SEPARATOR);
```

Comment:
Why not use split?

Reply:
I don't think the values are prohibited from using `=`, so it is more clear to slice before&after parts at the first `=` instead of doing a join on `parts[1...n]`

Reply:
Comments in the design spec state that '=' and ';' should never appear in the values :)

Reply:
Updated to use split